flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -247,27 +275,148 @@ public int compare(final StateTransition<T> o1, final StateTransition<T> o2) {
 	 * @return Collection of computation states which result from the current one
 	 */
 	private Collection<ComputationState<T>> computeNextStates(
-			final ComputationState<T> computationState,
-			final T event,
-			final long timestamp) {
-		Stack<State<T>> states = new Stack<>();
-		ArrayList<ComputationState<T>> resultingComputationStates = new ArrayList<>();
-		State<T> state = computationState.getState();
+		final ComputationState<T> computationState,
+		final T event,
+		final long timestamp) {
+		final ArrayList<ComputationState<T>> resultingComputationStates = new ArrayList<>();
```

Comment:
Move the `resultingComputationStates` just before the `for (StateTransition<T> edge : edges) {...}`.